src/App.test.tsx

```diff
@@ -30,4 +32,98 @@ describe('App component:', () => {
       expect(screen.getByText(/emotions category/i)).toBeInTheDocument();
     });
   });
+  // ***************************************************INTEGRATION TESTS**********************************************
+  describe('opens', () => {
+    it('sidebar on header button click', () => {
+      const renderComp = render(
+        <MemoryRouter initialEntries={['/']}>
+          <App />
+        </MemoryRouter>
+      );
+      const openButton = renderComp.container.getElementsByClassName('SideBarOpenButton');
+      // before user click
+      // now SideBar is not active
+      expect(renderComp.container.getElementsByClassName('SideBar active')[0]).toBeFalsy(); // is not active
+      // user click
+      userEvent.click(openButton[0]); // we have one element therefore index is 0
+      // "Sidebar active" is a correct classname when sidebar is open
+      expect(renderComp.container.getElementsByClassName('SideBar active')[0]).toBeInTheDocument(); // is active and visible
+    });
+
+    it('selected category on category cell click', () => {
+      const renderComp = render(
+        <MemoryRouter>
+          <App />
+        </MemoryRouter>
+      );
+      const elements = renderComp.container.getElementsByClassName('Link');
+      // before user click
+      expect(
+        renderComp.queryByText('Action (Set B) category. Feel free to practice!')
+      ).not.toBeInTheDocument();
+      // user click
+      userEvent.click(elements[1]); // click on action set A card
+      // Action set B is a correct page
+      expect(
+        renderComp.getByText('Action (Set B) category. Feel free to practice!')
+      ).toBeInTheDocument(); // correct page
+      expect(
+        renderComp.queryByText('Emotions category. Feel free to practice!')
+      ).not.toBeInTheDocument(); // incorrect page
+    });
+  });
+
+  describe('shows', () => {
+    it('start button if switch state changes', () => {
+      const renderComp = render(
+        <MemoryRouter initialEntries={['/category/1']}>
+          <App />
+        </MemoryRouter>
+      );
+      const elements = renderComp.container.getElementsByClassName('Switch'); // get switch element
+      // before user click
+      expect(
+        renderComp.container.getElementsByClassName('StartGameButton hide')[0]
+      ).toBeInTheDocument(); // check button hide class
+      // then get a part of switch to click
+      const switchChild = elements[0].childNodes[0] as Element;
```

Comment:
please find a way to avoid array like style requests. Is looks a bit like magic.